File: src/main/java/de/hsa/bpmnandai/project/api/executables/DmnExecutables.java
package de.hsa.bpmnandai.project.api.executables;

import de.hsa.bpmnandai.project.domain.facade.ExternalDmnFacade;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@Component
@RequiredArgsConstructor
public class DmnExecutables {

    private final ExternalDmnFacade externalDmnFacade;

    public String checkCreditworthiness(final String requestId) {
        log.info("Executing check");
        return externalDmnFacade.checkCreditworthiness(requestId);
    }

}

File: src/main/kotlin/de/emaarco/bpmnandai/project/api/executables/DmnExecutables.kt
package de.emaarco.bpmnandai.project.api.executables

import de.emaarco.bpmnandai.project.domain.facade.ExternalDmnFacade
import org.springframework.stereotype.Component

@Component
class DmnExecutables(private val externalDmnFacade: ExternalDmnFacade) {

    /**
     * A DMN-task that is invoked to check whether a customer is creditworthy
     * (...customer has created a credit-request that can be identified by the provided ID)
     */
    fun checkCreditworthiness(requestId: String): String {
        println("Check whether credit-application '${requestId}' can be approved")
        return externalDmnFacade.checkCreditworthiness(requestId)
    }

}